### PR TITLE
fix: return data from cheatcodes

### PR DESCRIPTION
**Pull request by aon**

### CODE REVIEW

File: e2e-tests/contracts/TestCheatcodes.sol [hunk 1]
```diff
@@ -1,8 +1,10 @@
 // SPDX-License-Identifier: MIT
 
 pragma solidity ^0.8.0;
+import "hardhat/console.sol";
 
 contract TestCheatcodes {
+  event LogNonce(uint64 data);
```
Comment (grw-ms) [inline e2e-tests/contracts/TestCheatcodes.sol]:
```suggestion

contract TestCheatcodes {
```

File: e2e-tests/contracts/TestCheatcodes.sol [hunk 2]
```diff
@@ -21,9 +23,19 @@ contract TestCheatcodes {
...
+      abi.encodeWithSignature("setNonce(address,uint64)", account, nonce)
+    );
     require(success, "setNonce failed");
+    (success, data) = CHEATCODE_ADDRESS.call(abi.encodeWithSignature("getNonce(address)", account));
+    require(success, "getNonce failed");
+//    emit LogNonce(data);
+     uint64 finalNonce = abi.decode(data, (uint64));
+//     emit LogNonce();
+     emit LogNonce(finalNonce);
+//    console.log(string(data));
```
Comment (grw-ms) [inline e2e-tests/contracts/TestCheatcodes.sol]:
```suggestion
     uint64 finalNonce = abi.decode(data, (uint64));
```

File: e2e-tests/test/cheatcodes.test.ts [hunk 3]
```diff
@@ -47,17 +47,20 @@ describe("Cheatcodes", function () {
...
-  it("Should test vm.setNonce", async function () {
+  it.only("Should test vm.setNonce and vm.getNonce", async function () {
     // Arrange
     const wallet = new Wallet(RichAccounts[0].PrivateKey);
     const deployer = new Deployer(hre, wallet);
     const randomWallet = Wallet.createRandom().connect(provider);
 
     // Act
     const cheatcodes = await deployContract(deployer, "TestCheatcodes", []);
+    console.log("cheatcodes", cheatcodes.address);
```
Comment (grw-ms) [inline e2e-tests/test/cheatcodes.test.ts]:
```suggestion
```

File: e2e-tests/test/cheatcodes.test.ts [hunk 4]
```diff
@@ -47,17 +47,20 @@ describe("Cheatcodes", function () {
...
+
     const initialNonce = await provider.getTransactionCount(randomWallet.address);
-    const tx = await cheatcodes.setNonce(randomWallet.address, 1234);
+    const tx = await cheatcodes.testGetSetNonce(randomWallet.address, 1234, { gasLimit: 1000000 });
     const receipt = await tx.wait();
+    console.log(receipt);
```
Comment (grw-ms) [inline e2e-tests/test/cheatcodes.test.ts]:
```suggestion
```

File: src/cheatcodes.rs [hunk 5]
```diff
@@ -47,21 +56,76 @@ abigen!(
...
+    ) {
+        // if self.returndata.is_some() {
+        // if let Opcode::UMA(a) = _data.opcode.variant.opcode {
+        //     if let UMAOpcode::HeapRead = a {
+        //         dbg!(_data);
+        //     }
+        // }
+        // }
+    }
+
```
Comment (grw-ms) [inline src/cheatcodes.rs]:
```suggestion
```

File: src/node/in_memory.rs [hunk 6]
```diff
@@ -1377,6 +1374,7 @@ impl<S: ForkSource + std::fmt::Debug + Clone + Send + Sync + 'static> InMemoryNo
         tracing::info!("│   TRANSACTION SUMMARY   │");
         tracing::info!("└─────────────────────────┘");
 
+        dbg!(&tx_result.result);
```
Comment (Jrigada) [inline src/node/in_memory.rs]: It should not be a problem in release but if it we do not need this statement we should remove it

File: e2e-tests/contracts/TestCheatcodes.sol (hunk 2 above)
Comment (aon) [reply e2e-tests/contracts/TestCheatcodes.sol]: Fixed in 8c46e906342d378104c7b3573fab7826891b8531

File: e2e-tests/test/cheatcodes.test.ts (hunk 3 above)
Comment (aon) [reply e2e-tests/test/cheatcodes.test.ts]: Fixed in 8c46e906342d378104c7b3573fab7826891b8531

File: src/cheatcodes.rs (hunk 5 above)
Comment (aon) [reply src/cheatcodes.rs]: Fixed in 8c46e906342d378104c7b3573fab7826891b8531

File: src/node/in_memory.rs (hunk 6 above)
Comment (aon) [reply src/node/in_memory.rs]: Fixed in 8c46e906342d378104c7b3573fab7826891b8531

**Pull request merged**